src/image/sample_image.cpp

```diff
@@ -0,0 +1,85 @@
+#include "sample_image.hpp"
+
+namespace lucidy{
+
+    SampleImage::SampleImage(){}
+
+    SampleImage::SampleImage(const char* path, settings::IMG::data & settings):
+        path(path),
+        settings(settings)
+    {init();}
+
+    void SampleImage::init(){
```

Comment:
Place the content of init in the constructor

Reply:
This makes code unnecessarily long and init function is alway more recommend. Invalid comment due to bad advice

Reply:
It does make a difference in performance. Since the `init()` function is virtual. Which means it has to be looked up at runtime what address the function is at. This consumes both flash memory for the V-Table and instructions when calling this function for dereferencing the function pointer. Placing the content of `init()` removes the necessity for a virtual function. 
It can be said that placing the content of init inside the constructor makes the code less maintainable, but that only really applies when you have a large `init()` function and more than one constructor calling `init()`
You can decide what to do, but I cannot assure that the development-to-master review will allow it.